sound/soc/soc-dapm.c

```diff
@@ -4464,9 +4464,17 @@ void snd_soc_dapm_connect_dai_link_widgets(struct snd_soc_card *card)
 			for_each_rtd_codec_dais(rtd, i, codec_dai)
 				dapm_connect_dai_pair(card, rtd, codec_dai,
 						      asoc_rtd_to_cpu(rtd, i));
+		} else if (rtd->dai_link->codec_ch_maps) {
+			int cpu_id;
+
+			for_each_rtd_codec_dais(rtd, i, codec_dai) {
+				cpu_id = rtd->dai_link->codec_ch_maps[i].connected_cpu_id;
+				dapm_connect_dai_pair(card, rtd, codec_dai,
+						      asoc_rtd_to_cpu(rtd, cpu_id));
+			}
```

Comment:
do we need to check that N < M here?

Reply:
> do we need to check that N < M here?

Yes, and I added a `if (cpu_id >= rtd->dai_link->num_cpus) ` test,